portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -1,1029 +1,982 @@
-			},
-			setDefaultOQL: function(alterations) {
-				config.default_oql = alterations;
-			}
-		};
-	})();
-	var objEach = function (iterable, callback) {
-		for (var k in iterable) {
-			if (iterable.hasOwnProperty(k)) {
-				callback(iterable[k], k);
-			}
+    var getCBioPortalMutationCounts = function (webservice_data) {
+	/* In: - webservice_data, a list of data obtained from the webservice API
+	 * Out: Map from gene+","+start_pos+","+end_pos to cbioportal mutation count for that position range and gene
+	 */
+	var counts_map = {};
+	var def = new $.Deferred();
+	var to_query = {};
+	for (var i = 0; i < webservice_data.length; i++) {
+	    var datum = webservice_data[i];
+	    if (datum.genetic_alteration_type !== "MUTATION_EXTENDED") {
+		continue;
+	    }
+	    var gene = datum.hugo_gene_symbol;
+	    var start_pos = datum.protein_start_position;
+	    var end_pos = datum.protein_end_position;
+	    if (gene && start_pos && end_pos && !isNaN(start_pos) && !isNaN(end_pos)) {
+		to_query[gene + ',' + parseInt(start_pos, 10) + ',' + parseInt(end_pos, 10)] = true;
+	    }
+	}
+	var queries = Object.keys(to_query).map(function (x) {
+	    var splitx = x.split(',');
+	    return {
+		gene: splitx[0],
+		start_pos: splitx[1],
+		end_pos: splitx[2]
+	    };
+	});
+	var genes = queries.map(function (q) {
+	    return q.gene;
+	});
+	var starts = queries.map(function (q) {
+	    return q.start_pos;
+	});
+	var ends = queries.map(function (q) {
+	    return q.end_pos;
+	});
+
+	window.cbioportal_client.getMutationCounts({
+	    'type': 'count',
+	    'per_study': false,
+	    'gene': genes,
+	    'start': starts,
+	    'end': ends,
+	    'echo': ['gene', 'start', 'end']
+	}).then(function (counts) {
+	    for (var i = 0; i < counts.length; i++) {
+		var gene = counts[i].gene;
+		var start = parseInt(counts[i].start, 10);
+		var end = parseInt(counts[i].end, 10);
+		counts_map[gene + ',' + start + ',' + end] = parseInt(counts[i].count, 10);
+	    }
+	    def.resolve(counts_map);
+	}).fail(function () {
+	    def.reject();
+	});
+	return def.promise();
+    };
+    var getOncoKBAnnotations = function(webservice_data) {
+	var is_oncogenic = {}; // #{gene}&#{alteration.toUpperCase()}&{tumor_type.toUpperCase()} is key, boolean is value
+	var query = {
+	    "geneStatus": "Complete",
+	    "source": "cbioportal",
+	    "evidenceTypes": "GENE_SUMMARY,GENE_BACKGROUND,ONCOGENIC,MUTATION_EFFECT,VUS,STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_SENSITIVITY,STANDARD_THERAPEUTIC_IMPLICATIONS_FOR_DRUG_RESISTANCE,INVESTIGATIONAL_THERAPEUTIC_IMPLICATIONS_DRUG_SENSITIVITY",
+	    "queries": [
+		{
+		    "hugoSymbol": "BRAF",
+		    "alteration": "V600E",
+		    "tumorType": "Melanoma"
 		}
-	};
-	var uniqStrings = function(array_of_strings) {
-	    var seen = {};
-	    var uniq = [];
-	    for (var i=0; i<array_of_strings.length; i++) {
-		var str = array_of_strings[i];
-		if (!seen[str]) {
-		    uniq.push(str);
-		    seen[str] = true;
+	    ],
```

Comment:
Very sorry, this accidentally got into this pull request in pre-finished form (can be ignored, is completed in a different branch). You're right though. This is the only thing that is unfinished in this pull request.